File: helfomat-infrastructure-jpa/src/main/java/de/helfenkannjeder/helfomat/infrastructure/jpa/EventRepository.java
package de.helfenkannjeder.helfomat.infrastructure.jpa;

import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

/**
 * @author Valentin Zickner
 */
public interface EventRepository extends JpaRepository<Event, EventId> {

    @Query("SELECT EP FROM Event EP ORDER BY EP.createdDate")
    List<Event> findAll();

    @Query("SELECT EP FROM Event EP WHERE EP.organizationId = :organizationId ORDER BY EP.createdDate")
    List<Event> findByOrganizationId(@Param("organizationId") OrganisationId organizationId);
}

File: helfomat-infrastructure-jpa/src/main/kotlin/de/helfenkannjeder/helfomat/infrastructure/jpa/EventRepository.kt
package de.helfenkannjeder.helfomat.infrastructure.jpa

import de.helfenkannjeder.helfomat.core.organization.OrganizationId
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

/**
 * @author Valentin Zickner
 */
interface EventRepository : JpaRepository<Event, EventId> {

    @Query("SELECT E FROM Event E ORDER BY E.createdDate")
    override fun findAll(): List<Event>

    @Query("SELECT E FROM Event E WHERE E.organizationId = :organizationId ORDER BY E.createdDate")
    fun findByOrganizationId(@Param("organizationId") organizationId: OrganizationId): List<Event>

}